### PR TITLE
WIP: DQ Tool from Feedback

### PR DESCRIPTION
**Pull request by awisema**
A first cut of a 'Client- Centric Punchlist'

### CODE REVIEW

File: drivers/hmis_data_quality_tool/app/controllers/hmis_data_quality_tool/warehouse_reports/reports_controller.rb [hunk 1]
```diff
@@ -36,6 +37,27 @@ def show
...
+          ::HmisDataQualityTool::Enrollment.sections(@report).map { |k, v| [k, v[:title]] } +
+          ::HmisDataQualityTool::CurrentLivingSituation.sections(@report).map { |k, v| [k, v[:title]] }
+      ).to_h
+
+      @flags = {}
+      @groups.values.map(&:keys).flatten.each do |key|
+        question_name = "#{@lookup[key]}__invalid"
+        @flags[@lookup[key]] = @report.universe(question_name).members.pluck(:personal_id, :data_source_id)
+      end
+      @clients_with_flags = @flags.values.flatten(1).uniq
```
Comment (eanders) [inline drivers/hmis_data_quality_tool/app/controllers/hmis_data_quality_tool/warehouse_reports/reports_controller.rb]: This all feels like something I'd like to move into the report model, so you could call:
```
@lookup = @report.pivot_lookup
@flags = @report.pivot_flags
@clients_with_flags  @report.pivot_clients_with_flags
```

Or better:
```
@pivot_details = @report.pivot_details
```
and just have it return a hash of what you need.

Comment (eanders) [reply drivers/hmis_data_quality_tool/app/controllers/hmis_data_quality_tool/warehouse_reports/reports_controller.rb]: I might be missing something, but I'm not seeing any compelling reason to have this logic in the controller.

File: drivers/hmis_data_quality_tool/app/views/hmis_data_quality_tool/warehouse_reports/reports/_by_client.haml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            %td.text-warning.icon-warning
+          -else
+            %td
+          %td= client.personal_id
+          %td= client.last_name
+          %td= client.first_name
+          - @groups.values.map(&:keys).flatten.each do |key|
+            - marked = @flags[@lookup[key]].include?([client.personal_id, client.data_source_id])
+            - if marked
+              %td.text-danger.text-center.icon-cross
```
Comment (eanders) [inline drivers/hmis_data_quality_tool/app/views/hmis_data_quality_tool/warehouse_reports/reports/_by_client.haml]:
```suggestion
              %td.text-center
                checkmark_or_x(false)
```
If we do that, we can easily add a tooltip later if we want, and swap between font and svg for the rendering if we need to.

**Review by eanders — APPROVED** [review]
Couple of requests, but this looks like a great start!  Thanks!

**Comment by awisema** [comment]
Cutting this part off from PDF and XLSX downloads.

**Review by eanders — APPROVED** [review]

**Pull request merged**